pkg/cli/mapping_server.go

```diff
@@ -117,15 +129,31 @@ func (c *MappingServerCommand) RunUnstarted(ctx context.Context, args []string)
 		return nil, nil, closer, fmt.Errorf("server.NewHandler: %w", err)
 	}
 
+	srv, err := serving.New(c.cfg.Port)
+	if err != nil {
+		return nil, nil, closer, fmt.Errorf("failed to create serving infrastructure: %w", err)
+	}
+
 	closer = func() {
-		if err := handler.Cleanup(); err != nil {
-			logger.Errorw("failed to close clean up handler", "error", err)
+		var retErr error
+		successTopic.Stop()
+		failureTopic.Stop()
+		if err := processorClient.Close(); err != nil {
+			retErr = errors.Join(retErr, fmt.Errorf("failed to stop processor: %w", err))
+		}
+		if retErr != nil {
+			logger.Errorw("failed to clean up resources", "error", err)
 		}
 	}
 
-	srv, err := serving.New(c.cfg.Port)
+	return srv, handler.HTTPHandler(), closer, nil
+}
+
+func createMessangerTopic(ctx context.Context, projectID, topicID string) (*pubsub.Topic, error) {
+	client, err := pubsub.NewClient(ctx, projectID)
```

Comment:
We can create a single pubsub client since we only have a single project. You can use a single client to create two topics. I think we can get rid of this func